x-pack/plugins/index_lifecycle_management/public/sections/edit_policy/components/node_allocation/node_allocation.js

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { Component, Fragment } from 'react';
+import { EuiSelect, EuiButtonEmpty, EuiCallOut, EuiSpacer, EuiLoadingSpinner } from '@elastic/eui';
+import { FormattedMessage, injectI18n } from '@kbn/i18n/react';
+import { PHASE_NODE_ATTRS } from '../../../../store/constants';
+import { ErrableFormRow } from '../../form_errors';
+import { LearnMoreLink } from '../../../components/learn_more_link';
+const learnMoreLinks = (
+  <Fragment>
+    <EuiSpacer size="s" />
+    <LearnMoreLink
+      docPath="indices-rollover-index.html"
+      text={
+        <FormattedMessage
+          id="xpack.indexLifecycleMgmt.editPolicy.learnAboutNodeAttributesLink"
+          defaultMessage="Learn about node attributes"
+        />
+      }
+    />
+    <EuiSpacer size="s" />
+    <LearnMoreLink
+      text={
+        <FormattedMessage
+          id="xpack.indexLifecycleMgmt.editPolicy.learnAboutShardAllocationLink"
+          defaultMessage="Learn about shard allocation"
+        />
+      }
+      docPath="shards-allocation.html"
+    />
+  </Fragment>
+);
+class NodeAllocationUi extends Component {
+  componentDidMount() {
+    this.props.fetchNodes();
+  }
+  render() {
+    const {
+      phase,
+      setPhaseData,
+      intl,
+      isShowingErrors,
+      phaseData,
+      showNodeDetailsFlyout,
+      nodeOptions,
+      errors
+    } = this.props;
+    if (!nodeOptions) {
```

Comment:
Looks like this branch never gets hit because `nodeOptions` is initialized as an empty array.